File: ARCADE/The Core/Level 6_1: isPower.cs
    using System;
     
    public class Test
    {
    	public static bool isPower(int n){
            if (n == 1) return true;
            for (int a = 2; a <= Math.Sqrt(n); a++){
                for (int b = 2; b <= Math.Sqrt(n); b++){
                    if (Math.Pow(a, b) == n) return true;
                }
            }
            return false;
        }
    	public static void Main()
    	{
    		int n = 125;
    		Console.WriteLine(isPower(n));
    		int m = 72;
    		Console.WriteLine(isPower(m));
    	}
    }

File: ARCADE/The Core/Level 6_1: isPower.pas
program isPower;
uses Math;

function isPower(n: integer): boolean;
var i, j : integer;
begin
    isPower := false;
    if(n = 1) then
    begin
        isPower := true;
        exit;
    end;
    for i := 1 to n do
        for j := 2 to n do
            if(Power(i, j) = n) then isPower := true;
end;

var n : integer;
begin
	n := 125;
	writeln(isPower(n));
	n := 72;
	writeln(isPower(n));
end.
